packages/moda-sdk/src/profile/updateProfile.ts

```diff
@@ -0,0 +1,40 @@
+import { Address, Hash } from 'viem';
+import { Config, ConfigError } from '../types';
+import { isValidAddress } from '../utils/isValidAddress';
+import { Result } from '../utils/types';
+import { verifyConfig } from '../utils/verifyConfig';
+import { ProfileABI } from './abi/ProfileABI';
+import { ProfileError } from './types';
+
+interface UpdateProfileArgs {
+  config: Config;
+  profileAddress: Address;
+  metadataUri: string;
+}
+
+export async function updateProfile(
+  updateProfileArgs: UpdateProfileArgs,
+): Promise<Result<Hash, ProfileError | ConfigError>> {
+  const { config, profileAddress, metadataUri } = updateProfileArgs;
+
+  const verifiedConfig = verifyConfig(config);
+  if (!verifiedConfig.ok) return verifiedConfig;
+  const { publicClient, walletClient } = verifiedConfig.value;
+
+  if (!isValidAddress(profileAddress))
+    return { ok: false, error: ProfileError.AddressInvalid };
+
+  const [account] = await walletClient.getAddresses();
+
+  const { request } = await publicClient.simulateContract({
+    account,
+    abi: ProfileABI,
+    address: profileAddress,
+    functionName: 'updateProfile',
+    args: [metadataUri],
+  });
+
+  const txHash = await walletClient.writeContract(request);
```

Comment:
What happens if this reverts? Is there a way we can return the reverted message as an error? Is this error detectable on line 29?

We may need to `try/catch` inside this function then return the error. It may also make sense to make the error type for this method a string so arbitrary messages can be returned from a contract. Providing enum errors are still useful for the ones we know